#6334 - Support Elasticsearch 1.0+

sevein opened the pull request:
Upgraded Elastica to v1.3.0.0 and addressed the incompatibility issues with the new API:
- `text query` replaced with `match query` (cb98f15)
- New `fields` structure in search respones (1c21987)
- Deprecated `multi_field` type (7a0238d)
- Standard analyzer with `_none_` stopwords (ab23b4a)

sevein committed:
Upgrade Elastica to v1.3.0.0

sevein committed:
Replace deprecated Query\Text with Query\Match

ES 1.0.0.RC1 removed the text query in support of the match query.
The behaviour of the match query is pretty similar to its predecessor.

sevein committed:
Update code to expect arrays where "fields" is used

Field values, in response to the fields parameter, are now always returned
as arrays by Elasticsearch. We don't really use "fields" much, but I have
to update the search/autocomplete template to extract the fields attending
to its new structure.

More info in "breaking changes in 1.0 » return values":
http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/_return_values.html

sevein committed:
Stop using multi_field type

Use the new syntax instead, introduced in ES 1.x.

sevein committed:
Add custom analyzers with stopwords lists

Before Elasticsearch 1.0, the standard analyzer used the list of English
stopwords by default. But now is set to use the empty stopwords.

This commit defines a set of new analyzers based on the standard analyzer
for each language where Elasticsearch provides a predefined stopwords list.

MikeFE commented:
Hi Jesús, it looks good to me. Would you like me to merge this, or did you want to?

sevein commented:
Oh, no worries. I'll do it. I want to test this with Dan before. Thanks.

sevein committed:
Fix indexing of lat/lon (geo_point type)

Nulls are not accepted, which is good. But this never came up in ES 0.90,
I'm guessing they've change they have improved the validation.

sevein committed:
Check that we are running a recent ES release

Added a constant MIN_VERSION = "1.3.0". Used version_compare() in order to
validate the release and cached the result with a short ttl to avoid hitting
the server every single request.
